builtin/logical/pki/path_tidy.go

```diff
@@ -358,13 +358,23 @@ func (b *backend) tidyStatusIncCertStoreCount() {
 	defer b.tidyStatusLock.Unlock()
 
 	b.tidyStatus.certStoreDeletedCount++
+
+	b.certsCountedLock.RLock()
+	defer b.certsCountedLock.RUnlock()
+
+	b.decrementTotalCertificatesCount()
 }
 
 func (b *backend) tidyStatusIncRevokedCertCount() {
 	b.tidyStatusLock.Lock()
 	defer b.tidyStatusLock.Unlock()
 
 	b.tidyStatus.revokedCertDeletedCount++
+
+	b.certsCountedLock.RLock()
```

Comment:
I'm a bit puzzled by these locking calls here and above in `tidyStatusIncCertStoreCount`? We aren't updating storage, or we have already, and the atomic operations should cover us for the inc/dec of the counter?

Reply:
This lock was basically to avoid writing to the certstore, while the initial count (list operation) was being performed.

Resolving this conversation, since we've decided no more lock.